tests/test_utils.py

```diff
@@ -88,9 +89,32 @@ def test_balance_unbalanced_data():
 
     assert(('data1', 1) == next(balanced_data))
     assert(('data4', 0) == next(balanced_data))
-    assert(('data3', 1) == next(balanced_data))
+    assert(('data2', 1) == next(balanced_data))
     assert(('data5', 0) == next(balanced_data))
+    assert(('data3', 1) == next(balanced_data))
+
+    with pytest.raises(StopIteration):
+        next(balanced_data)
+
```

Comment:
Add the iterator test in the same function, so you don't have to define the data twice. You could also reuse the assertions if you do something like this:
```Python
for input_type in [list, iter]:
    balanced_data = utils.balance(input_type(unbalanced_tuples))
    ...
```

Reply:
There should be even a pytest way to parametrize the function, if you want.